### PR TITLE
Mimites, final fixes(hopefully) - Requested by EvilDragon

### PR DESCRIPTION
**Pull request by ro5490**
## About The Pull Request

Last minute fixes that should have been on the previous PR before merging

add Undef
moves removal from list to Destroy()
moves stuck threshold to define

## Why It's Good For The Game

See above

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl:
code: Threshold Define, and add destroy for Mimites
/:cl:

### CODE REVIEW

File: code/modules/events/mimite_infestation.dm [hunk 1]
```diff
@@ -58,7 +58,7 @@
 
 
 /datum/round_event/mimite_infestation/end()
-	if(LAZYLEN(GLOB.all_mimites) >= 1) //If they're still around at the end of 60 min, turn of replication permanently
+	if(length(GLOB.all_mimites)) //If they're still around at the end of 60 min, turn of replication permanently
```
Comment (PowerfulBacon) [inline code/modules/events/mimite_infestation.dm:61]: If all mimites is a lazy list, then it should be using LAZYLEN. If it isn't, then you shouldn't be using LAZYADD

Comment (ro5490) [reply code/modules/events/mimite_infestation.dm:61]: Tis a list, was suggested to do this here - https://github.com/BeeStation/BeeStation-Hornet/pull/11194#discussion_r1682049951

Comment (EvilDragonfiend) [reply code/modules/events/mimite_infestation.dm:61]: ah, yes, I misdirected him to use `length()`. it should be lazylen.

Comment (ro5490) [reply code/modules/events/mimite_infestation.dm:61]: will revert

**Review by EvilDragonfiend — APPROVED** [review]
yep, it's alright now.

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**